### PR TITLE
Update brand container to use latest psammead-brand version

### LINKED ISSUE #1805: Update brand component to the latest version.
**Is your feature request related to a problem? Please describe.**

Following [Psammead#528](https://github.com/bbc/psammead/pull/528) and [Psammead#480](https://github.com/bbc/psammead/pull/480) `Brand` now allows for scaling of the brand svg  and dynamic `link`. Simorgh still runs version `1.0.1` while `Brand` is at version `3.0.0`

When the brand is clicked it should take the user to the website home page (in the front page case it's the page the user is already on): 

Igbo: https://www.bbc.com/igbo
Yoruba: https://www.bbc.com/yoruba
Pidgin: https://www.bbc.com/pidgin

**Describe the solution you'd like**
Update `@bbc/psammead-brand` to latest version and add `links` to service config so that it can be passed via `ServiceContext`

**Describe alternatives you've considered**
N/A

**Testing notes**
[Tester to complete]

Dev insight: Will Cypress tests be required or are unit tests sufficient? Will there be any potential regression? etc

**Additional context**

### PR DESCRIPTION
**Pull request by Bopchy**
Resolves #1805 

**Overall change:** _Update brand container to use latest psammead-brand component._

**Code changes:**

- _Update psammead-brand dependency to ^4.0.0_
- _Add product, serviceLocalizedName, svgHeight, minWidth, maxWidth, url props to brand container._
- _Add relative service links to service configs._
- _Update tests and snapshots._
- _Update storybook._
---

- [x] I have assigned myself to this PR and the corresponding issues
- [ ] Tests added for new features
- [ ] Test engineer approval
- [ ] I have followed [the merging checklist](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/MERGE_PROCESS.md) and this is ready to merge.

### CODE REVIEW

File: src/app/lib/config/services/igbo.js
Comment (ghost) [inline src/app/lib/config/services/igbo.js]: Do we need to add `svgRatio` to config?

**Review by ghost — CHANGES_REQUESTED** [review]

File: src/app/lib/config/services/igbo.js
Comment (Bopchy) [reply src/app/lib/config/services/igbo.js]: We actually don't have to - have removed svgRatio from config. svgRatio is now being retrieved from brandSvg

File: src/app/containers/Brand/index.jsx
Comment (Bopchy) [inline src/app/containers/Brand/index.jsx:12]: Added this check because test withPageWrapper.test.jsx was failing, and seeing brandSVG as undefined despite defining it in

Comment (ghost) [reply src/app/containers/Brand/index.jsx:12]: I don't think we need that check either. You could pass `service` to  `<PageWrapperHOC />`. `withPageWrapper` is failing because no service was passed to `PageWrapperHOC` it therefore defaults to the default service config. The `default` config doesn't have `brandSVG` defined. `configs` are in `lib/services`

Comment (Bopchy) [reply src/app/containers/Brand/index.jsx:12]: Hmmh, the news `service` is currently being passed to `<PageWrapperHOC />` through dataProps,  but (without the check to see if brandSVG is defined) - the test still fails

Comment (ghost) [reply src/app/containers/Brand/index.jsx:12]: You need to pass it on the component `<PageWrapperHOC data={dataProps} service="news" />`

Comment (ghost) [reply src/app/containers/Brand/index.jsx:12]: `withPageWrapper` returns `defaultPageWrapper` which expects service as a prop.

Comment (Bopchy) [reply src/app/containers/Brand/index.jsx:12]: 👍 Made the change

File: src/app/lib/config/services/news.js
Comment (jamesbhobbs) [inline src/app/lib/config/services/news.js]: needs www. also I'm unsure if we should can/should use .co.uk or .com hardcoded or pass it in conditionally based on the variant we're rendering. Suggest you check with other maintainers.

Comment (Bopchy) [reply src/app/lib/config/services/news.js]: Checked and @pjlee11 supports having the option of both .co.uk and .com

This will be handled in #1909

File: src/app/lib/config/services/igbo.js
Comment (ghost) [inline src/app/lib/config/services/igbo.js]: Could we change this to something more descriptive. ie. `frontpageUrl`

**Review by ghost — CHANGES_REQUESTED** [review]

**Review by ghost — DISMISSED** [review]
Looks great. 👍

**Comment by DenisBBC** [comment]
Is it worth to wait until [this](https://github.com/bbc/psammead/pull/609) PR is merged, before moving this forward? Otherwise, we will have to update again the container.

**Review by DenisHdz — COMMENTED** [review]
LGTM, but do you think is it worth to wait until [this](https://github.com/bbc/psammead/pull/609) PR is merged, before moving this forward? Otherwise, we will have to update the container again.

**Comment by Bopchy** [comment]
Yes, let's wait until it's merged

File: src/app/containers/Brand/index.jsx
Comment (12) [inline src/app/containers/Brand/index.jsx]: Just a thought, do we need to define a default `brandSVG` here:

https://github.com/bbc/simorgh/blob/f745b0082b783ebd27fc2eee57bf6ce6138d6ffe/src/app/lib/config/services/index.js#L13-L22

as we have this that sets the default service to `default`:

https://github.com/bbc/simorgh/blob/f745b0082b783ebd27fc2eee57bf6ce6138d6ffe/src/app/contexts/ServiceContext/index.jsx#L19-L21

Comment (12) [reply src/app/containers/Brand/index.jsx]: I realise this isn't a change you've explicitly made, however it now fails more abruptly now when you don't filter out the `default` story. 

(also I don't see the point in having a default value if we're just going to filter it out in storybook 🤷‍♂)

Currently, `latest` throws a failed prop type, however still renders the component, without the SVG.
This branch would now throw a `TypeError` which causes the entire component to fail and not render.

I think this should be addressed before merging

**Review by 12 — DISMISSED** [review]

File: src/app/containers/Brand/index.jsx
Comment (Bopchy) [reply src/app/containers/Brand/index.jsx]: Handled failure in absence of brandSVG by adding a check for brandSVG here - https://github.com/bbc/simorgh/blob/28dec513a1bf6fd2e7043e9eaf63c4b6c53ddc1b/src/app/containers/Brand/index.jsx#L11. 

Regarding if a default brandSVG should be defined - it's been decided it's not necessary, because (with the added check) the component works without an SVG, and the prop errors are currently not an issue

**Comment by Bopchy** [comment]
Waiting on https://github.com/bbc/psammead/pull/609 to be merged so that changes are made once

**Comment by dr3** [comment]
This will now pass the URL link as `/news` or `/igbo` etc, to be relative.

**Comment by Bopchy** [comment]
Now waiting on https://github.com/bbc/psammead/issues/606

UPDATE: Doesn't need to wait on  https://github.com/bbc/psammead/issues/606 as Simorgh will pick up the latest non-major change in the psammead-brand package

File: src/app/lib/config/services/news.js
Comment (Bopchy) [inline src/app/lib/config/services/news.js:11]: Added this because Storybook fails on the news service without it. I am not sure if it is required

**Review by j-pendlebury — APPROVED** [review]
LGTM 👍

**Review by dr3 — CHANGES_REQUESTED** [review]
This change makes the brand not be restricted to 1280px, is it broken in psammead?
image

**Comment by dr3** [comment]
This PR is blocked on https://github.com/bbc/psammead/issues/645

File: cypress/integration/frontpageLayoutSpec.js
Comment (Bopchy) [inline cypress/integration/frontpageLayoutSpec.js]: Using `${serviceConfig.url}` in favour of ` ${config.baseUrl}${serviceConfig.url}` since BrandContainer passes a relative url

File: package.json
Comment (dr3) [inline package.json]: Needs to be ^4.0.1

File: src/app/containers/Brand/index.jsx
Comment (dr3) [inline src/app/containers/Brand/index.jsx:17]: Need to pass `borderTop` or `borderBottom` depending on if its in the header or footer, can be set by passing a prop when calling this container

File: src/app/containers/PageHandlers/withPageWrapper.test.jsx
Comment (dr3) [inline src/app/containers/PageHandlers/withPageWrapper.test.jsx]: why this change? I dont think service is an expected prop https://github.com/bbc/simorgh/blob/1805-update-brand-component-to-latest-version/src/app/containers/PageHandlers/withPageWrapper.jsx#L14

File: src/app/containers/Brand/index.jsx
Comment (dr3) [inline src/app/containers/Brand/index.jsx]: Can we just use `/${service}`, saves us adding a new item to the config

**Review by dr3 — DISMISSED** [review]

File: src/app/containers/PageHandlers/withPageWrapper.test.jsx
Comment (Bopchy) [reply src/app/containers/PageHandlers/withPageWrapper.test.jsx]: Added because `PageWrapperHOC` uses the `default` service, and not the news service specified in dataProps - unless the service is passed to it as a prop

Comment (Bopchy) [reply src/app/containers/PageHandlers/withPageWrapper.test.jsx]: This change isn't needed. Reverted it.

File: src/app/containers/Brand/index.jsx
Comment (ghost) [inline src/app/containers/Brand/index.jsx:24]: Could we make this a variable.

File: src/app/containers/Header/__snapshots__/index.test.jsx.snap
Comment (ghost) [inline src/app/containers/Header/__snapshots__/index.test.jsx.snap]: Why are we getting undefined here?

**Review by ghost — CHANGES_REQUESTED** [review]

File: src/app/containers/Header/__snapshots__/index.test.jsx.snap
Comment (Bopchy) [reply src/app/containers/Header/__snapshots__/index.test.jsx.snap]: This was because the source of the url wasn't defined in the test stub. This is now defined

**Review by ghost — CHANGES_REQUESTED** [review]
Fix conflict

File: cypress/integration/frontpageLayoutSpec.js
Comment (ryanmccombe) [inline cypress/integration/frontpageLayoutSpec.js]: `.should('have.attr', 'href', serviceConfig.url)`

File: cypress/integration/frontpageLayoutSpec.js
Comment (ryanmccombe) [inline cypress/integration/frontpageLayoutSpec.js]:
```suggestion
            .should('have.attr', 'href', serviceConfig.url)
```

File: cypress/integration/frontpageLayoutSpec.js
Comment (ryanmccombe) [inline cypress/integration/frontpageLayoutSpec.js]:
```suggestion
            .should('have.attr', 'href', serviceConfig.url)
```

**Review by ryanmccombe — APPROVED** [review]

File: src/app/lib/config/services/news.js
Comment (ryanmccombe) [reply src/app/lib/config/services/news.js:11]: I think the issue is Brand container is now expecting a key called `product` on context, where before it was using `brandName` (the value on the previous line)

You may also have to make this change to the other languages, or have Brand container continue to use `brandName` from the ServiceContext, instead of this new `product` value

Comment (Bopchy) [reply src/app/lib/config/services/news.js:11]: Aah, yes. I thought there might be a reason why it was added to other services but not to news.

**Review by ghost — APPROVED** [review]
LGTM 👍

**Comment by david-boydell** [comment]
LGTM this is ready for merge

File: src/app/containers/Header/index.jsx
Comment (jroebu14) [inline src/app/containers/Header/index.jsx:6]: I'm a bit late with this one but wouldn't it be better to just do `<BrandContainer borderBottom />` here and `<BrandContainer borderTop />` in the footer and then in `Brand` you can spread the props on the component like

```jsx
    <Brand
      product={product}
      serviceLocalisedName={serviceLocalizedName}
      svgHeight={svgMaxHeight}
      minWidth={minWidth}
      maxWidth={maxWidth}
      svg={brandSVG}
      url={`/${service}`}
      {...props}
    />
```

Comment (ghost) [reply src/app/containers/Header/index.jsx:6]: @jroebu14 Not too late. Made that change  https://github.com/bbc/simorgh/pull/2146

**Pull request merged**
